Source/Charts/Data/Implementations/Standard/ChartDataSet.swift

```diff
@@ -278,6 +278,7 @@ open class ChartDataSet: ChartBaseDataSet
         closestToY yValue: Double,
         rounding: ChartDataSetRounding) -> Int
     {
+        guard xValue.isFinite else { return -1 }
```

Comment:
why guard inf here? I was thinking you are talking about NaN here, but

Reply:
It is an attempt to validate input before it causes the library to enter an infinite loop. From my observations NaN values cause this behavior.

Reply:
why you get NaN here?

Reply:
NaN is the default x value for PieChartDataEntry

Reply:
where it goes into a dead loop? Currently pie chart data entry is working fine I suppose in ChartsDemo and our CI tests. We need to understand where this lead to a dead loop.